src/main/kotlin/com/terraformation/backend/customer/api/SubLocationsController.kt

```diff
@@ -0,0 +1,141 @@
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@CustomerEndpoint
+@RestController
+@RequestMapping("/api/v1/facilities/{facilityId}/subLocations")
+class SubLocationsController(
+    private val accessionStore: AccessionStore,
+    private val facilityStore: FacilityStore,
+) {
+  @GetMapping
+  @Operation(summary = "Gets a list of sub-locations at a facility.")
+  fun listSubLocations(@PathVariable facilityId: FacilityId): ListSubLocationsResponsePayload {
+    val locations = facilityStore.fetchSubLocations(facilityId)
+    val accessionCounts = accessionStore.countActiveBySubLocation(facilityId)
+
+    return ListSubLocationsResponsePayload(
+        locations.map { SubLocationPayload(it, accessionCounts[it.id!!], null) })
+  }
+
+  @GetMapping("/{subLocationId}")
+  @Operation(summary = "Gets information about a specific sub-location at a facility.")
+  fun getSubLocation(
+      @PathVariable facilityId: FacilityId,
+      @PathVariable subLocationId: SubLocationId
+  ): GetSubLocationResponsePayload {
+    val location = facilityStore.fetchSubLocation(subLocationId)
+    val accessionCount = accessionStore.countActiveInSubLocation(subLocationId)
+
+    return GetSubLocationResponsePayload(location, accessionCount, null)
+  }
+
+  @ApiResponse200
+  @ApiResponse409(
+      description = "A sub-location with the requested name already exists at the facility.")
+  @Operation(summary = "Creates a new sub-location at a facility.")
+  @PostMapping
+  fun createSubLocation(
+      @PathVariable facilityId: FacilityId,
+      @RequestBody payload: CreateSubLocationRequestPayload,
+  ): GetSubLocationResponsePayload {
+    val facility = facilityStore.fetchOneById(facilityId)
+    val id = facilityStore.createSubLocation(facilityId, payload.name)
+    val accessionCount = if (facility.type == FacilityType.SeedBank) 0 else null
+
+    return GetSubLocationResponsePayload(
+        SubLocationPayload(
+            activeAccessions = accessionCount,
+            facilityId = facilityId,
+            id = id,
+            name = payload.name))
+  }
+
+  @ApiResponse200
+  @ApiResponse409(
+      description = "A sub-location with the requested name already exists at the facility.")
+  @Operation(summary = "Updates the name of a sub-location at a facility.")
+  @PutMapping("/{subLocationId}")
+  fun updateSubLocation(
+      @PathVariable facilityId: FacilityId,
+      @PathVariable subLocationId: SubLocationId,
+      @RequestBody payload: UpdateSubLocationRequestPayload,
+  ): SimpleSuccessResponsePayload {
+    val location = facilityStore.fetchSubLocation(subLocationId)
+    if (location.facilityId != facilityId) {
+      throw SubLocationNotFoundException(subLocationId)
+    }
+
+    facilityStore.updateSubLocation(subLocationId, payload.name)
+
+    return SimpleSuccessResponsePayload()
+  }
+}
```

Comment:
delete sublocation endpoint?

Reply:
> delete sublocation endpoint?

I left it out because deleting sub-locations isn't in the requirements and would need extra product definition: what happens if there are seedlings assigned to the sub-location, what happens to historical records of changes to sub-location, etc..

Reply:
Oh, you're right, though. We don't need deletion for nursery sub-locations, but we do need it if we want to switch the seed bank storage locations code to these new endpoints. Will add it in a followup PR.